app/coffee/Features/Authorization/AuthorizationManager.coffee

```diff
@@ -0,0 +1,61 @@
+CollaboratorsHandler = require("../Collaborators/CollaboratorsHandler")
+Project = require("../../models/Project").Project
+User = require("../../models/User").User
+
+module.exports = AuthorizationManager =
+	# Get the privilege level that the user has for the project
+	# Returns:
+	#	* privilegeLevel: "owner", "readAndWrite", of "readOnly" if the user has
+	#	  access. false if the user does not have access
+	#   * becausePublic: true if the access level is only because the project is public.
+	getPrivilegeLevelForProject: (user_id, project_id, callback = (error, privilegeLevel, becausePublic) ->) ->
+		getPublicAccessLevel = () ->
+			Project.findOne { _id: project_id }, { publicAccesLevel: 1 }, (error, project) ->
+				return callback(error) if error?
+				if project.publicAccesLevel in ["readOnly", "readAndWrite"]
+					return callback null, project.publicAccesLevel, true
+				else
+					return callback null, false, false
+		
+		if !user_id?
+			getPublicAccessLevel()
+		else
+			CollaboratorsHandler.getMemberIdPrivilegeLevel user_id, project_id, (error, privilegeLevel) ->
+				return callback(error) if error?
+				if privilegeLevel? and privilegeLevel
```

Comment:
Yes. I wanted to make it explicit that we're doing a null check first, and then explicitly testing the criteria of privilegeLevel == false indicating no access (as opposed to null which would indicate something had gone wrong). It's a bit convoluted though, I agree.